### PR TITLE
Creating "ask new question" button in "explore page"

### LINKED ISSUE #51: Update links in Explore-page
- Add "add question" button that leads to "add question page" (should be synchronized with @avivz450 )

- Click on question in explore page leads to the selected question page (should be synchronized  with @liornoy )

### PR DESCRIPTION
**Pull request by danitLevi**
- Creating "ask new question" button in explore page . Clicking on it leads to "new question" page
- Testing "ask new qiestion" btn :
  - Asserting that the response content contains the url of "new question"   page

fix #51

### CODE REVIEW

**Review by ifireball — CHANGES_REQUESTED** [review]
This is a huge PR - please break it apart. Usally when you see you're getting a single PR that fixes more then one issue - its a good indicator you've included too much stuff in the same PR.

**Review by avivz450 — COMMENTED** [review]
I think you should edit the models.py and add min and max value to the book_page field in the "Question" model. That way the user will not be able to filter a book that his page is negative, or too high.
You can do it by adding `from django.core.validators import MinValueValidator, MaxValueValidator` at the beginning of models.py and by adding ` validators=[MinValueValidator(1), MaxValueValidator(999)]` inside models.IntegerField in book_page field.

File: home/models.py [hunk 1]
```diff
@@ -88,6 +88,9 @@ class Meta:
     def __str__(self):
         return f"Question #{self.id} : {self.title} ( {self.publish_date.strftime('%d/%m/%Y %H:%M')} )"
 
+    def get_answers_num(self):
+        return self.answer_set.count()
+
```
Comment (liornoy) [inline home/models.py:93]: What is answer_set? we don't have that field in Question model

File: home/templates/home/explore.html [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+.landing-bg{
+	background-color: #f2dccb;
+}
+
+.collapsed-bg
+{
+    background-color: #f3ebe6;
+}
+
+</style>
```
Comment (liornoy) [inline home/templates/home/explore.html:31]: I wrote all the style in the HTML also, but Barak suggest that we move it do a different CSS file in the Static folder

File: .idea/.gitignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Default ignored files
```
Comment (liornoy) [inline .idea/.gitignore:1]: What are these files in .idea folder are for?

**Review by liornoy — COMMENTED** [review]
I saw some things to comment about but I think in order to really understand the changes, you should squash all the commits so the HTML file will be complete

File: .idea/.gitignore (hunk 3 above)
Comment (danitLevi) [reply .idea/.gitignore:1]: @liornoy 
I usually use PyCharm and it automatically creates "idea" folder that contains settings for the project. I accidentally added it to the PR. I will remove it and add it to the git ignore so it won't happen again.

File: home/models.py (hunk 1 above)
Comment (danitLevi) [reply home/models.py:93]: Yes, We don't have answer_set field in Question model .

The _set suffix is a reverse lookup class variable in Django.
You can read more about it in the following link https://docs.djangoproject.com/en/3.2/topics/db/examples/many_to_one/

You're welcome to ask me if you have any further questions regarding this issue .

File: home/templates/home/explore.html (hunk 2 above)
Comment (danitLevi) [reply home/templates/home/explore.html]: I moved it to different css file as requested

**Comment by danitLevi** [comment]
> I think you should edit the models.py and add min and max value to the book_page field in the "Question" model. That way the user will not be able to filter a book that his page is negative, or too high.
> You can do it by adding `from django.core.validators import MinValueValidator, MaxValueValidator` at the beginning of models.py and by adding ` validators=[MinValueValidator(1), MaxValueValidator(999)]` inside models.IntegerField in book_page field.


@avivz450 
I add the min and max value to the book_page field in the filter form. 
(I didn't change the question model as you suggested because it didn't work for filtering, but I change the filter set I created so the user will not be able to filter book page by negative or too high value)

**Comment by danitLevi** [comment]
> This is a huge PR - please break it apart. Usally when you see you're getting a single PR that fixes more then one issue - its a good indicator you've included too much stuff in the same PR.

Hi @ifireball 

At first, I installed Pytest in this PR and added "pytest.ini" and ".github/workflows/pytest.yml" files. I removed them now in order to prevent conflicts (I saw Ido already added them in the PR he created )

Besides, I usually use PyCharm and it automatically creates an "idea" folder that contains settings for the project. It seems that initially I accidentally added it to the PR, and maybe it causes the PR to look huge.

Moreover, I moved the tests I created to the "explore page" into different PR. 

Is this PR OK now? Or is it still too big?
I know this PR handling multiple issues but all this issues are related to the same page (explore page ) and each commit is based on the previous commit . Should I separate them into separated PRs?

In addition, I have a question. Before changing this PR I created PR #75 on top of it (because I needed the URL of explore page in this PR). Therefore the commits from this PR also appear in PR #75, but now the commits in PR #75 didn't change automatically according to the changes I made.  Can it cause problems after merging this PR ? Should do anything else in PR #75 in this case?

**Comment by danitLevi** [comment]
> I saw some things to comment about but I think in order to really understand the changes, you should squash all the commits so the HTML file will be complete

@liornoy  You can see the final files by clicking on the "fiels changed"

**Comment by ifireball** [comment]
@danitLevi about the size of the PR - the issues you've created actually represent a good division of topics - each one of them should be its own PR, with its own tests.

Doing it like that means that once you wrote a PR for issue #48 you can do what you needed to do in PR #75 without having to implement the rest.

This might mean you may not finish implementing everything you have here for the demo but that is fine, its a good lesson about how much we think we need to implement v.s. how much we actually need.

File: home/models.py (hunk 1 above)
Comment (liornoy) [reply home/models.py:93]: Cool didn't know it. Thanks!!

**Comment by danitLevi** [comment]
> @danitLevi about the size of the PR - the issues you've created actually represent a good division of topics - each one of them should be its own PR, with its own tests.
> 
> Doing it like that means that once you wrote a PR for issue #48 you can do what you needed to do in PR #75 without having to implement the rest.
> 
> This might mean you may not finish implementing everything you have here for the demo but that is fine, its a good lesson about how much we think we need to implement v.s. how much we actually need.

@ifireball 
I split my PR into several PRs as you requested.
For each PR I added the commits it depends on. Did I create them ok?
You can review them in the following order :
PR #80  --- >  PR #75  --- >  PR #81 --- >  PR #82 --- >  PR #83 --- >  PR #74

**Comment by danitLevi** [comment]
@ifireball  @AviadP

This PR is ready for review

**Comment by rebeccaTubman** [comment]
Looking good

**Comment by ifireball** [comment]
Team: can you please request and do review properly and not just comment?

**Review by ifireball — APPROVED** [review]
LGTM but missing team reviews

**Comment by rebeccaTubman** [comment]
> Looking good

I think this can be merged.

**Comment by ifireball** [comment]
@rebeccaTubman why are you not posting a review approval?

File: home/templates/home/explore.html [hunk 4]
```diff
@@ -10,6 +10,13 @@
 <form method="GET" class="container">
 <div class="m-5" >
 
+    <div class="row mb-4">
```
Comment (rebeccaTubman) [inline home/templates/home/explore.html:13]: A great PR with a small feature with test coverage.

**Review by rebeccaTubman — APPROVED** [review]

**Review by liornoy — APPROVED** [review]

**Review by avivz450 — APPROVED** [review]

**Pull request merged**